### PR TITLE
Issue #3000444: Better geocoder service api refactoring for 3.x branch

### PR DESCRIPTION
**Pull request by itamair**
This do corresponds to the last Merged PR (https://github.com/drupol/geocoder/pull/22), for the 3.x branch.

### CODE REVIEW

File: src/Geocoder.php [hunk 1]
```diff
@@ -2,13 +2,23 @@
...
 class Geocoder implements GeocoderInterface {
 
+  /**
+   * The config factory service.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $config;
+
+
```
Comment (drupol) [inline src/Geocoder.php]: Let's keep 1 single empty lines instead of 2.

File: src/Geocoder.php [hunk 2]
```diff
@@ -19,21 +29,33 @@ class Geocoder implements GeocoderInterface {
...
+  public function __construct(ConfigFactoryInterface $config_factory, ProviderPluginManager $provider_plugin_manager) {
+    $this->config = $config_factory->get('geocoder.settings');
     $this->providerPluginManager = $provider_plugin_manager;
   }
 
   /**
    * {@inheritdoc}
    */
   public function geocode($data, array $plugins, array $options = []) {
+
```
Comment (drupol) [inline src/Geocoder.php]: Let's remove those empty lines at the beginning of methods.

File: src/Geocoder.php [hunk 3]
```diff
@@ -53,10 +75,19 @@ public function geocode($data, array $plugins, array $options = []) {
    * {@inheritdoc}
    */
   public function reverse($latitude, $longitude, array $plugins, array $options = []) {
+
```
Comment (drupol) [inline src/Geocoder.php]: Same here.

File: src/Geocoder.php (hunk 1 above)
Comment (itamair) [reply src/Geocoder.php]: Should be fixed

File: src/Geocoder.php (hunk 2 above)
Comment (itamair) [reply src/Geocoder.php]: Should be fixed

File: src/Geocoder.php (hunk 3 above)
Comment (itamair) [reply src/Geocoder.php]: Should be fixed

**Comment by itamair** [comment]
I accomplished in the last commit all the requested changes

**Pull request merged**